pkg/policy/api/l4.go

```diff
@@ -28,6 +28,12 @@ func (l4 L4Proto) IsAny() bool {
 	return l4 == ProtoAny || string(l4) == ""
 }
 
+// SupportedProtocols returns the currently supported protocols in the policy
+// engine, excluding "ANY".
+func SupportedProtocols() []L4Proto {
+	return []L4Proto{ProtoTCP, ProtoUDP, ProtoSCTP}
+}
```

Comment:
Out of curiosity, why are ICMP{,v6} protocols not included here? That seems to contradict the comment. More in general, any reason not to use `ANY` as protocol in the deny rules in lockdown mode?

Reply:
We do not support selecting ICMP ([there is a way to do it](https://docs.cilium.io/en/latest/security/policy/language/#limit-icmp-icmpv6-types)). Only these protocols are supported for network policy selection when using the general PortoProtocol type.

Reply:
Sure. But I guess you may want to also block ICMP/ICMP6 traffic when entering in lockdown mode, otherwise traffic that was previously possibly blocked by the policy would then be allowed.

Reply:
No it won't. Explicit deny-entries, counter-intuitively, deny all other traffic as well implicitly. ICMP logic is separate from this.

Reply:
Ah gotcha. Thinking a bit more about this, I guess that even an empty map should do the trick (without the explicit denies), but those make it more explicit that traffic will be denied.

Reply:
> I guess that even an empty map should do the trick (without the explicit denies), but those make it more explicit that traffic will be denied.

We would need at least one deny-entry, but correct.

Reply:
I guess that's because the userspace logic inserts the allow all entry inside the map if no explicit allow/deny entries are present. Otherwise at the BPF level an empty map should drop everything IIUC.

Reply:
>  Otherwise at the BPF level an empty map should drop everything IIUC.

That depends on a number of config options. We want the explicitly entries.